si/session.py

```diff
@@ -137,37 +141,107 @@ def decide(self, decision_name, candidates, **options):
         used to send a decide event to SI servers, and to make a decision
         this method does not throw an exception
         """
-        return self.decide_internal(decision_name=decision_name, context_name="", candidates=candidates,
-                                    with_context=False, properties=None, options=options)
+        return self.decide_internal(decision_name=decision_name, context_name="",
+                                    candidates=candidates, with_context=False, properties=None,
+                                    options=options)
 
     def decide_with_context(self, context_name, candidates, decision_name, properties, **options):
         """
         used to send a decide with context request to SI servers, and to make a decision
         this method does not throw an exception
         """
-        return self.decide_internal(decision_name=decision_name, context_name=context_name, candidates=candidates,
-                                    with_context=True, properties=properties, options=options)
+        return self.decide_internal(decision_name=decision_name, context_name=context_name,
+                                    candidates=candidates, with_context=True, properties=properties,
+                                    options=options)
+
+    def multi_decide_with_context(self, context_name: str, properties, decisions, **options):
+        logger: logging._loggerClass = options.get('logger', self.logger)
+        reply = {"ampToken": '', "decisions": []}
+        for decision_request in decisions:
+            candidates = decision_request["candidates"]
+            reply["decisions"].append({"decision": get_default_candidate(candidates),
+                                       "fallback": True,
+                                       "failureReason": "Another decision request failed"})
+        if not context_name or not self.user_id or not self.session_id:
+            reply['error'] = 'missing context_name, userId or sessionId'
+            logger.error(reply['error'])
+            return reply
+        request = {
+            "name": context_name,
+            "properties": properties,
+            "decisionNames": [],
+            "decisions": [],
+            "userId": self.user_id,
+            "sessionId": self.session_id,
+            "index": self.atomic_next_index(),
+            "ts": int((time.time() + 0.5) * 1000),
```

Comment:
fyi: if this is trying to round to the nearest millisecond, it should do something else. As it is, it's adding 500 (+ - 1) milliseconds to the current time. Not a major problem, but might not be the intended effect